Fix Promise Race Condition When Testing Solutions

This issue highlights a problem in the `testSolutions` function. The issue arises when calls to the `testCppSolution` function throw an error before entering the for loop that awaits the promises from that function. This issue can be observed by modifying the sample solutions to cause them to fail. The log output below shows where the crash occurs:

```
✖ Failed to test sample/problems/andBooleans/solution.cpp
  EEXIST: file already exists, mkdir 'build/sample/problems/andBooleans'

✖ Failed to test sample/problems/appendIntegersArrays/solution.cpp
  Command failed: clang++ --std=c++20 -O2 build/sample/problems/appendIntegersArrays/test.cpp -o build/sample/problems/appendIntegersArrays/test
  In file included from build/sample/problems/appendIntegersArrays/test.cpp:1:
  build/sample/problems/appendIntegersArrays/../../../../sample/problems/appendIntegersArrays/solution.cpp:6:59: error: use of undeclared identifier 'asdf'
      nums1.insert(nums1.end(), nums2.begin(), nums2.end());asdf
                                                            ^
  1 error generated.

⠸ Testing sample/problems/appendStrings/solution.cpp...node:internal/errors:984
  const err = new Error(message);
              ^

Error: Command failed: build/sample/problems/multiplyIntegers/test
failed to test example 1:
.  output: 13
.  expected: 40

failed to test example 2:
.  output: -4
.  expected: -32

failed to test example 3:
.  output: 1000
.  expected: 0

3 test cases have failed

    at genericNodeError (node:internal/errors:984:15)
    at wrappedFn (node:internal/errors:538:14)
    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:519:28)
    at ChildProcess.emit (node:domain:488:12)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:519:28)
    at Socket.emit (node:domain:488:12)
    at Pipe.<anonymous> (node:net:337:12) {
  code: 3,
  killed: false,
  signal: null,
  cmd: 'build/sample/problems/multiplyIntegers/test',
  stdout: 'testing example 1...\ntesting example 2...\ntesting example 3...\n',
  stderr: 'failed to test example 1:\n' +
    '.  output: 13\n' +
    '.  expected: 40\n' +
    '\n' +
    'failed to test example 2:\n' +
    '.  output: -4\n' +
    '.  expected: -32\n' +
    '\n' +
    'failed to test example 3:\n' +
    '.  output: 1000\n' +
    '.  expected: 0\n' +
    '\n' +
    '3 test cases have failed\n'
}

Node.js v21.6.1
```

--- a/src/test/index.ts
+++ b/src/test/index.ts
@@ -11,20 +11,26 @@ import { testCppSolution } from "./cpp/index.js";
 export async function testSolutions(solutionFiles: string[]): Promise<number> {
   const tests = solutionFiles.map((file) => ({
     file: file,
-    prom: testCppSolution(file),
+    prom: (async () => {
+      try {
+        await testCppSolution(file);
+      } catch (err) {
+        return err;
+      }
+    })(),
   }));
 
   let failures = 0;
   const spinner = ora();
   for (const test of tests) {
     spinner.start(`Testing ${test.file}...`);
-    try {
-      await test.prom;
-      spinner.succeed(`Tested ${test.file}`);
-    } catch (err) {
+    const err = await test.prom;
+    if (err) {
       ++failures;
       const msg = getErrorMessage(err).replaceAll("\n", "\n  ").trimEnd();
       spinner.fail(`Failed to test ${test.file}\n  ${msg}\n`);
+    } else {
+      spinner.succeed(`Tested ${test.file}`);
     }
   }
 

